crates/secret-service-proto/src/v1/wire.rs

```diff
@@ -0,0 +1,244 @@
+use bitcoin::{
+    hashes::Hash,
+    taproot::{ControlBlock, TaprootError},
+    ScriptBuf, TapNodeHash,
+};
+use musig2::errors::{RoundContributionError, RoundFinalizeError};
+use rkyv::{with::Map, Archive, Deserialize, Serialize};
+use strata_bridge_primitives::scripts::taproot::TaprootWitness;
+
+use super::traits::{Musig2SessionId, SignerIdxOutOfBounds};
+
+#[derive(Debug, Clone, Archive, Serialize, Deserialize)]
+pub enum ServerMessage {
+    InvalidClientMessage,
+    OpaqueServerError,
+
+    OperatorSignPsbt {
+        sig: [u8; 64],
+    },
+    OperatorPubkey {
+        pubkey: [u8; 33],
+    },
+
+    SignP2P {
+        sig: [u8; 64],
+    },
+    P2PPubkey {
+        pubkey: [u8; 33],
+    },
+
+    Musig2NewSession(Result<Musig2SessionId, SignerIdxOutOfBounds>),
+    Musig2Pubkey {
+        pubkey: [u8; 33],
+    },
+
+    Musig2FirstRoundOurNonce {
+        our_nonce: [u8; 66],
+    },
+    Musig2FirstRoundHoldouts {
+        pubkeys: Vec<[u8; 33]>,
+    },
+    Musig2FirstRoundIsComplete {
+        complete: bool,
+    },
+    Musig2FirstRoundReceivePubNonce(
+        #[rkyv(with = Map<super::rkyv_wrappers::RoundContributionError>)]
+        Option<RoundContributionError>,
+    ),
+    Musig2FirstRoundFinalize(
+        #[rkyv(with = Map<super::rkyv_wrappers::RoundFinalizeError>)] Option<RoundFinalizeError>,
+    ),
+
+    Musig2SecondRoundAggNonce {
+        nonce: [u8; 66],
+    },
+    Musig2SecondRoundHoldouts {
+        pubkeys: Vec<[u8; 33]>,
+    },
+    Musig2SecondRoundOurSignature {
+        sig: [u8; 32],
+    },
+    Musig2SecondRoundIsComplete {
+        complete: bool,
+    },
+    Musig2SecondRoundReceiveSignature(
+        #[rkyv(with = Map<super::rkyv_wrappers::RoundContributionError>)]
+        Option<RoundContributionError>,
+    ),
+    Musig2SecondRoundFinalize(Musig2SessionResult),
+
+    WotsGet160Key {
+        key: [u8; 20 * 160],
+    },
+    WotsGet256Key {
+        key: [u8; 20 * 256],
+    },
```

Comment:
Why these are `Get`?
```suggestion
    Wots160Key {
        key: [u8; 20 * 160],
    },
    Wots256Key {
        key: [u8; 20 * 256],
    },
```
Maybe `s/Key/{Pub,Sec}Key`